### PR TITLE
Fix Remove-Module tests where it fails on subsequent runs

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

On Windows, tests are effectively run twice (elevated and non-elevated).  In this case, the Remove-Module tests fail the second time because a module is set to `Constant` which means it can't be removed and the property can't be set again (even to `Constant`).  So that throws an exception during test setup and causes a bunch of failures.

Fix is to detect that the const module is already loaded and do nothing.  Also needed to rename the const module so other tests don't accidentally detect it on subsequent runs.

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Comment by PoshChan** [comment]
@SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-macos`
Send-MailMessage.Error occurred in Describe block
```powershell
Cannot bind argument to parameter 'Path' because it is null.
At /Users/vsts/agent/2.150.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1:11 char:30
```

**Comment by PoshChan** [comment]
@SteveL-MSFT, your last commit had 7 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Dot-source operator is not allowed in modules on locked down systems that export functions with wildcards.Verifies that a classes only module with dot-source and with using directive loads successfully
```powershell
The term 'PublicUsingFn' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Security\ConstrainedLanguageModules.Tests.ps1: line 426
426:                 $result = PublicUsingFn

```
Import trusted manifest files in system lockdown mode.Verifies that trusted manifest with nested modules and no extension module exports explicit function
```powershell
Cannot load the module 'C:\Users\VSSADM~1\AppData\Local\Temp\65cdf662-4b96-42a2-8527-f2cafdbaf743\ImportTrustedManifestWithNestedModsAndNoExtNoFnExport_System32' because the module nesting limit has been exceeded. Modules can only be nested to 10 levels. Evaluate and change the order in which you are loading modules to prevent exceeding the nesting limit, and then try running your script again.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Security\ConstrainedLanguageModules.Tests.ps1: line 1214
1214:                 $module = Import-Module -Name $manifestFileName -Force -PassThru

```
PowerShellGet - Module tests (Admin).Error occurred in Describe block
```powershell
Exception calling "ShouldContinue" with "2" argument(s): "Object reference not set to an instance of an object."
At D:\a\1\s\src\powershell-win-core\bin\release\netcoreapp3.0\win7-x64\publish\Modules\PowerShellGet\PSModule.psm1:11044 char:9
```
PowerShellGet - Script tests (Admin).Error occurred in Describe block
```powershell
Exception calling "ShouldContinue" with "2" argument(s): "Object reference not set to an instance of an object."
At D:\a\1\s\src\powershell-win-core\bin\release\netcoreapp3.0\win7-x64\publish\Modules\PowerShellGet\PSModule.psm1:11044 char:9
```
Handling of globbing patterns.Handling of Unix [ab] globbing patterns in literal paths.Rename-Item -LiteralPath can rename a file with Unix globbing characters
```powershell
Cannot rename the specified target, because it represents a path or device name.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystem.Tests.ps1: line 365
365:             Rename-Item -LiteralPath $file.FullName -NewName $newPath

```

**Comment by PoshChan** [comment]
@SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-linux`
Send-MailMessage.Error occurred in Describe block
```powershell
Cannot bind argument to parameter 'Path' because it is null.
At /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1:11 char:30
```

**Comment by PoshChan** [comment]
@SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-macos`
Send-MailMessage.Error occurred in Describe block
```powershell
Cannot bind argument to parameter 'Path' because it is null.
At /Users/vsts/agent/2.150.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1:11 char:30
```

**Comment by PoshChan** [comment]
@SteveL-MSFT, your last commit had 7 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Dot-source operator is not allowed in modules on locked down systems that export functions with wildcards.Verifies that a classes only module with dot-source and with using directive loads successfully
```powershell
The term 'PublicUsingFn' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Security\ConstrainedLanguageModules.Tests.ps1: line 426
426:                 $result = PublicUsingFn

```
Import trusted manifest files in system lockdown mode.Verifies that trusted manifest with nested modules and no extension module exports explicit function
```powershell
Cannot load the module 'C:\Users\VSSADM~1\AppData\Local\Temp\560fb1c0-81a8-4373-8c80-9595c47625ef\ImportTrustedManifestWithNestedModsAndNoExtNoFnExport_System32' because the module nesting limit has been exceeded. Modules can only be nested to 10 levels. Evaluate and change the order in which you are loading modules to prevent exceeding the nesting limit, and then try running your script again.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Security\ConstrainedLanguageModules.Tests.ps1: line 1214
1214:                 $module = Import-Module -Name $manifestFileName -Force -PassThru

```
PowerShellGet - Module tests (Admin).Error occurred in Describe block
```powershell
Exception calling "ShouldContinue" with "2" argument(s): "Object reference not set to an instance of an object."
At D:\a\1\s\src\powershell-win-core\bin\release\netcoreapp3.0\win7-x64\publish\Modules\PowerShellGet\PSModule.psm1:11044 char:9
```
PowerShellGet - Script tests (Admin).Error occurred in Describe block
```powershell
Exception calling "ShouldContinue" with "2" argument(s): "Object reference not set to an instance of an object."
At D:\a\1\s\src\powershell-win-core\bin\release\netcoreapp3.0\win7-x64\publish\Modules\PowerShellGet\PSModule.psm1:11044 char:9
```
Handling of globbing patterns.Handling of Unix [ab] globbing patterns in literal paths.Rename-Item -LiteralPath can rename a file with Unix globbing characters
```powershell
Cannot rename the specified target, because it represents a path or device name.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystem.Tests.ps1: line 365
365:             Rename-Item -LiteralPath $file.FullName -NewName $newPath

```

**Comment by PoshChan** [comment]
@SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-linux`
Send-MailMessage.Error occurred in Describe block
```powershell
Cannot bind argument to parameter 'Path' because it is null.
At /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1:11 char:30
```

File: test/powershell/Modules/Microsoft.PowerShell.Core/Remove-Module.Tests.ps1 [hunk 1]
```diff
@@ -264,8 +264,15 @@ Describe "Remove-Module : module is readOnly | Constant" -Tags "CI" {
...
         Import-Module -Name "$testdrive\Modules\Foo\Foo_rw.psd1" -Force
         Import-Module -Name "$testdrive\Modules\Bar\Bar_rw.psd1" -Force
-        Import-Module -Name "$testdrive\Modules\Baz\Baz_const.psd1" -Force
-        (Get-Module -Name "Baz_const").AccessMode = "Constant"
+
+        # If this test is run again in the same session, this test module
+        # will already be loaded and can't be removed because it sets
+        # AccessMode to Constant, so only import this test module and
+        # set AccessMode if it isn't already loaded
+        if ($null -eq (Get-Module "Const_module")) {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Remove-Module.Tests.ps1]:
```suggestion
        if ($null -eq (Get-Module -Name "Const_module")) {
```

**Review by iSazonov — APPROVED** [review]
LGTM with one minor comment.

**Comment by pougetat** [comment]
Sorry about this by the way (I introduced this issue with my remove-module tests)

**Comment by iSazonov** [comment]
@pougetat No problem. Iterations during development are common.

**Comment by SteveL-MSFT** [comment]
@pougetat I appreciate your contributions.  We all make mistakes.  I don't know why the tests weren't failing previously.

**Pull request merged**